File: Trimestre 5/API REST/src/models/entrada.model.js
import { Model, DataTypes } from "sequelize";
import sequelize from "../config/connect.db.js";
import Producto from "./producto.model.js";
import Proveedor from "./proveedor.model.js";
import Persona from "./persona.model.js";

class Entrada extends Model {}

Entrada.init(
  {
    idEntrada: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
    },
    Fecha: {
      type: DataTypes.DATE,
      allowNull: false,
    },
    Cantidad: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    Producto_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: Producto,
        key: "idProducto",
      },
    },
    Proveedor_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: Proveedor,
        key: "idProveedor",
      },
    },
    Persona_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: Persona,
        key: "idPersona",
      },
    },
  },
  {
    sequelize,
    modelName: "Entrada",
    tableName: "Entradas",
    timestamps: false,
  }
);

export default Entrada;

File: Trimestre 6/FluttertablaEntradas/Flutter/frontend/lib/models/entrada_model.dart
class Entrada {
  final int? idEntrada;
  final String fecha;
  final String descripcion;
  final int cantidad;
  final int productoId;

  Entrada({
    this.idEntrada,
    required this.fecha,
    required this.descripcion,
    required this.cantidad,
    required this.productoId,
  });

  factory Entrada.fromJson(Map<String, dynamic> json) {
    return Entrada(
      idEntrada: json['idEntrada'],
      fecha: json['fecha'],
      descripcion: json['descripcion'],
      cantidad: json['cantidad'],
      productoId: json['producto_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'fecha': fecha,
      'descripcion': descripcion,
      'cantidad': cantidad,
      'producto_id': productoId,
    };
  }
}
